source/atomics.tex

```diff
@@ -1503,6 +1508,723 @@
+\indexlibrarymember{operator=}{atomic_ref<T*>}%
+\indexlibrarymember{operator=}{atomic_ref<\placeholder{integral}>}%
+\indexlibrarymember{operator=}{atomic_ref<\placeholder{floating-point}>}%
+\begin{itemdecl}
+T operator=(T desired) const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+store(desired);
+return desired;
+\end{codeblock}
+\end{itemdescr}
+
+\indexlibrarymember{operator \placeholder{type}}{atomic_ref}%
+\indexlibrarymember{operator T*}{atomic_ref<T*>}%
+\indexlibrarymember{operator \placeholder{integral}}{atomic_ref<\placeholder{integral}>}%
+\indexlibrarymember{operator \placeholder{floating-point}}{atomic_ref<\placeholder{floating-point}>}%
+\begin{itemdecl}
+operator T() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to: \tcode{return load();}
+\end{itemdescr}
+
+\indexlibrarymember{is_lock_free}{atomic_ref}%
+\indexlibrarymember{is_lock_free}{atomic_ref<T*>}%
+\indexlibrarymember{is_lock_free}{atomic_ref<\placeholder{integral}>}%
+\indexlibrarymember{is_lock_free}{atomic_ref<\placeholder{floating-point}>}%
+\begin{itemdecl}
+bool is_lock_free() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns \tcode{true} if the object's operations are lock-free,
+\tcode{false} otherwise.
+\end{itemdescr}
+
+\indexlibrarymember{store}{atomic_ref}%
+\indexlibrarymember{store}{atomic_ref<T*>}%
+\indexlibrarymember{store}{atomic_ref<\placeholder{integral}>}%
+\indexlibrarymember{store}{atomic_ref<\placeholder{floating-point}>}%
+\begin{itemdecl}
+void store(T desired, memory_order order = memory_order_seq_cst) const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\requires The \tcode{order} argument shall not be
+\tcode{memory_order_consume},
+\tcode{memory_order_acquire}, nor
+\tcode{memory_order_acq_rel}.
+
+\pnum
+\effects Atomically replaces the value referenced by \tcode{*ptr}
+with the value of desired.
+Memory is affected according to the value of \tcode{order}.
+\end{itemdescr}
+
+\indexlibrarymember{load}{atomic_ref}%
+\indexlibrarymember{load}{atomic_ref<T*>}%
+\indexlibrarymember{load}{atomic_ref<\placeholder{integral}>}%
+\indexlibrarymember{load}{atomic_ref<\placeholder{floating-point}>}%
+\begin{itemdecl}
+T load(memory_order order = memory_order_seq_cst) const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\requires The \tcode{order} argument shall not be
+\tcode{memory_order_release} nor \tcode{memory_order_acq_rel}.
+
+\pnum
+\effects Memory is affected according to the value of \tcode{order}.
+
+\pnum
+\returns Atomically returns the value referenced by \tcode{*ptr}.
+\end{itemdescr}
+
+\indexlibrarymember{exchange}{atomic_ref}%
+\indexlibrarymember{exchange}{atomic_ref<T*>}%
+\indexlibrarymember{exchange}{atomic_ref<\placeholder{integral}>}%
+\indexlibrarymember{exchange}{atomic_ref<\placeholder{floating-point}>}%
+\begin{itemdecl}
+T exchange(T desired, memory_order order = memory_order_seq_cst) noexcept;
```

Comment:
The paper is missing a `const` here, presumably a typo (since it's in the class definition).

Reply:
Fixed.